rules/force_load_direct_deps.bzl

```diff
@@ -1,36 +1,114 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain", "use_cpp_toolchain")
 load("//rules:providers.bzl", "AvoidDepsInfo")
 load("//rules:transition_support.bzl", "transition_support")
 
-def _impl(ctx):
-    if not ctx.attr.should_force_load:
-        return apple_common.new_objc_provider()
+def _objc_provider_static_libraries(dep):
+    """Returns the ObjcProvider static libraries (.a) that should be force loaded.
+    """
+    if not apple_common.Objc in dep:
+        return []
+
+    return dep[apple_common.Objc].library.to_list()
+
+def _cc_info_static_libraries(dep):
+    """Returns the CcInfo static libraries (.a) that should be force loaded.
+
+    NOTE: CcInfo, unlike ObjcProvider, does not encode where the static library came from.
+    In the existing `_objc_provider_static_libraries` we only collect `.library` from ObjcProvider.
+    These include only .a library files that are direct deps of the target, and NOT imported static libraries.
+    CcInfo only provides `.static_library` and does not make this distinction, we infer it here based on the path.
+    """
+    if not CcInfo in dep:
+        return []
+
+    static_cc_libraries = []
+    for linker_input in dep[CcInfo].linking_context.linker_inputs.to_list():
+        for library_to_link in linker_input.libraries:
+            if not library_to_link.static_library:
+                continue
+            containing_path = paths.dirname(library_to_link.static_library.path)
+            if containing_path.endswith(".framework"):
+                continue
```

Comment:
Can you add a comment here or in the doc comment what/why we're attempting to exclude here?  Is having the `.framework` suffix a proxy for "an imported static library" that isn't a direct dep?

Reply:
Updated the doc comment to go into more details